### PR TITLE
Hot tanks

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Touching a tank hotter than 80 °C will provide burns to your hand, you'll need proper equipment to handle hot tanks (protecting gloves)

## Why It's Good For The Game
More uses for protecting gloves (such as black gloves)

## Changelog
:cl:
balance: tanks with content over 80 °C will burn your hand if not protected by gloves; you can find them at atmos and toxins or you can buy them at cargo
/:cl:

### CODE REVIEW

**Comment by InsaneRed** [comment]
does insuls protect from the heat

File: code/game/objects/items/tanks/tanks.dm [hunk 1]
```diff
@@ -339,4 +339,41 @@
...
+	var/celsius_temperature = KELVIN_TO_CELSIUS(air_contents.temperature)
+
+	// make it burn hands unless you're wearing heat insulated gloves or have the RESISTHEAT/RESISTHEATHANDS traits (recycled comment as it still fits)
+	if(celsius_temperature >= 50)
+		var/protection = FALSE
+		var/mob/living/carbon/human/human = user
+		if(human.gloves)
+			var/obj/item/clothing/gloves/gloves = human.gloves
+			if(gloves.max_heat_protection_temperature)
+				protection = (gloves.max_heat_protection_temperature > 360)
```
Comment (NotRanged) [inline code/game/objects/items/tanks/tanks.dm:353]: protection = TRUE and merge the conditional into the if() above perhaps?

**Comment by Sparkezel** [comment]
Make CE gloves good at handling temperature and give atmos techs black gloves from the start in that case, otherwise imo I think it might make stuff a tidbit too tedious

**Comment by Ghilker** [comment]
> Make CE gloves good at handling temperature and give atmos techs black gloves from the start in that case, otherwise imo I think it might make stuff a tidbit too tedious

atmos has black gloves in their department

**Comment by joookschad** [comment]
how is this balance this is adding a useless chore lol

**Comment by maxymax13** [comment]
> how is this balance this is adding a pointless chore lol

ever tried lugging around a tank full of pressurized super hot death gas

**Comment by KathyRyals** [comment]
It makes sense, but also I'm really not sure about the benefits this potentially brings ? How high can the damage go ? Do you think it makes for a potentially more fun gameplay loop for atmos and science ? It's not forget that toxins handles very hot tanks as well and they don't have gloves in there.

**Comment by moocowswag** [comment]
I take it scientists will need to get telekinesis before they are allowed to do toxins because of this?

**Comment by Ghilker** [comment]
I've added 2 black gloves to toxins and now they are available to be bought at cargo

File: code/game/objects/items/tanks/tanks.dm (hunk 1 above)
Comment (Ghilker) [reply code/game/objects/items/tanks/tanks.dm:353]: this allow different levels of heat protection (max_heat_protection_temperature could be less than 360 in theory)

**Comment by joookschad** [comment]
> > how is this balance this is adding a pointless chore lol
> 
> ever tried lugging around a tank full of pressurized super hot death gas



ever tried making a tank full of pressurized super hot death gas? no because these generally dont exist irl and this is a game

**Comment by SkeletalElite** [comment]
How fast can this kill? Is it possible that someone could kill people by putting tanks in their hands?

**Comment by optimumtact** [comment]
> 
> 
> does insuls protect from the heat

he tried it

**Comment by OmegaDarkPotato** [comment]
So if I'm reading the code right, which I may not be, (feel free to call me a dumbass in such case) you're giving someone a critical burn wound and slapping em with 15 burn damage for picking up a tank that's over 1000c.
This can be done with even a T3 heater.

Critical burn wounds are an absolute pain to deal with as a medical doctor, and the only time I've ever seen em in practice was when someone died on a bonfire and stayed there for 2 minutes heating up to 9999K, or when Cimika spawned a couple of death squaddies with instructions to pulse rifle the crew for fun.

considering that someone could literally just drop a couple emergency oxygen tanks heated to 1000C with incredible ease over the hallways and then spike anyone who dares to check them out with critical burn wounds this seems a little over-tuned.

At the very least, I'd think that if you want to add some form of realism in terms of "tanks containing superheated gas are painful to touch", you should either A) drop the wound or B) have some sort of visual indicator that they're clearly going to hurt you if you try and pick them up

**Comment by CRITAWAKETS** [comment]
> So if I'm reading the code right, which I may not be, (feel free to call me a dumbass in such case) you're giving someone a critical burn wound and slapping em with 15 burn damage for picking up a tank that's over 1000c.
> This can be done with even a T3 heater.
> 
> Critical burn wounds are an absolute pain to deal with as a medical doctor, and the only time I've ever seen em in practice was when someone died on a bonfire and stayed there for 2 minutes heating up to 9999K, or when Cimika spawned a couple of death squaddies with instructions to pulse rifle the crew for fun.
> 
> considering that someone could literally just drop a couple emergency oxygen tanks heated to 1000C with incredible ease over the hallways and then spike anyone who dares to check them out with critical burn wounds this seems a little over-tuned.
> 
> At the very least, I'd think that if you want to add some form of realism in terms of "tanks containing superheated gas are painful to touch", you should either A) drop the wound or B) have some sort of visual indicator that they're clearly going to hurt you if you try and pick them up

you can literally just examine the tanks and it will tell you if they're hot

**Comment by SuperNovaa41** [comment]
Would black gloves fully protect you from any burn chance? Even 20000k+?

**Comment by Ghilker** [comment]
> Would black gloves fully protect you from any burn chance? Even 20000k+?

yes

**Comment by maxymax13** [comment]
make them glow red

**Comment by tommysalami3** [comment]
You should probably add a visual cue for when it reaches the temp where it can burn you, like a red tone over the sprite. It's not necessary, but it would help indicate the change in behavior. Pretty sure this is what maxy means when he says make them glow red, too.

**Comment by GunHog** [comment]
If I were to slip a superheated emergency tank (the small ones) into someone's pocket, would it burn him when he tries to remove it?

**Comment by Ghilker** [comment]
> If I were to slip a superheated emergency tank (the small ones) into someone's pocket, would it burn him when he tries to remove it?

i suppose so yes

**Comment by Ghilker** [comment]
now when you try to take a hot tank in your hand you'll drop it on the floor (so that it prevents impossible gimmiks like the pocket one)

**Comment by Ziiro** [comment]
>considering that someone could literally just drop a couple emergency oxygen tanks heated to 1000C with incredible ease over the hallways and then spike anyone who dares to check them out with critical burn wounds this seems a little over-tuned.

image

File: code/modules/cargo/goodies.dm [hunk 2]
```diff
@@ -79,6 +79,12 @@
 	cost = PAYCHECK_MEDIUM * 8
 	contains = list(/obj/item/clothing/gloves/color/yellow)
 
+/datum/supply_pack/goody/heat_resistant_gloves
+	name = "Heat resistant gloves Single-Pack"
+	desc = "A spare pair of heat resistant gloves, useful for handling hot stuff (clown doesn't count)."
```
Comment (maxymax13) [inline code/modules/cargo/goodies.dm:84]: 😳

File: code/game/objects/items/tanks/tanks.dm [hunk 3]
```diff
@@ -339,4 +341,42 @@
...
+				if(251 to 500)
+					wound_type = /datum/wound/burn/severe
+					heat_damage = 10
+				if(80 to 250)
+					wound_type = /datum/wound/burn/moderate
+					heat_damage = 5
+			affecting?.force_wound_upwards(wound_type)
+			if(affecting?.receive_damage( 0, heat_damage ))
+				human.update_damage_overlays()
+			to_chat(user, "<span class='warning'>You burn your hand on the tank!</span>")
```
Comment (maxymax13) [inline code/game/objects/items/tanks/tanks.dm:377]: make it have a different message when you drop it at 50-79 c

Comment (Ghilker) [reply code/game/objects/items/tanks/tanks.dm:377]: you don't drop it at 50-79 °C, you pick it up and have a custom message for it
```
if(protection || HAS_TRAIT(user, TRAIT_RESISTHEAT) || HAS_TRAIT(user, TRAIT_RESISTHEATHANDS) || celsius_temperature < 80)
	to_chat(user, "<span class='notice'>The tank feels very hot.</span>")```

**Comment by moocowswag** [comment]
Do lizards take reduced damage since they are all about reduced damage from heat?

**Comment by SkeletalElite** [comment]
Maybe considering lowering wound tiers a little bit
ie. no  wound for the first tier
level 1 wound for 2nd tier 
level 2 wound for 3rd.

Getting a severe burn wound because someone put a hot tank in your pocket just doesn't seem great to me

**Comment by anconfuzedrock** [comment]
Is this implying these tanks are completely unsafe and in no way insulated, so they should just cool off? Does this _actually_ make sense? Because it definitely doesn't make the game better.

**Comment by Kurgis** [comment]
If tanks are no longer fully insulated, would there be future plans for extremely hot tanks to radiate heat into the room they are in? It just seems odd that the "heat" only affects mobs on touch, but otherwise remain static in the environment they lay within.

**Comment by FernandoJ8** [comment]
If plasmamen engineers get their envirogloves insulated, plasmamen atmos techs should get this heat protection in their gloves, no?

**Comment by antropod** [comment]
> 
> 
> I take it scientists will need to get telekinesis before they are allowed to do toxins because of this?
Should burn your brain if you touch it with TK, like SM dusts your brain if you do that

**Pull request closed without merging**